packages/edit-site/src/components/post-edit/index.js

```diff
@@ -46,9 +46,9 @@ function PostEditForm( { postType, postId } ) {
 	const { saveEntityRecord } = useDispatch( coreDataStore );
 	const { fields } = usePostFields();
 	const form = {
-		visibleFields: [ 'title', 'author' ],
+		visibleFields: [ 'title', 'author', 'date' ],
 	};
-	const [ edits, setEdits ] = useState( {} );
+	const [ edits, setEdits ] = useState( initialEdits );
```

Comment:
This makes sure that the dataform has access to the previous item. It was always an empty object for the `onChange` event handler. The `setEdits( {} )` below also triggered a re-render of the DataForm on submit.